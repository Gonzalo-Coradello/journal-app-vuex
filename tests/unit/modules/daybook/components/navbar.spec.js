import Navbar from '@/modules/daybook/components/Navbar.vue'
import { shallowMount } from '@vue/test-utils'
import { beforeEach, describe, expect, test, vi } from 'vitest'
import createVuexStore from '../../../mock-data/mock-store'

describe('Navbar component', () => {
  const store = createVuexStore({
    user: {
      name: 'Juan Carlos',
      email: 'juan@gmail.com'
    },
    status: 'authenticated',
    idToken: 'ABC',
    refreshToken: 'XYZ'
  })

  beforeEach(() => vi.clearAllMocks())

  test('debe mostrar el componente correctamente', () => {
    const wrapper = shallowMount(Navbar, {
      global: {
        plugins: [store]
      }
    })

    expect(wrapper.html).toMatchSnapshot()
  })

  test('click en logout debe cerrar sesiÃ³n y redireccionar', async () => {
    const wrapper = shallowMount(Navbar, {
      global: {
        plugins: [store]
      }
    })

    await wrapper.find('button').trigger('click')

    expect(wrapper.router.push).toHaveBeenCalledWith({ name: 'login' })

    expect(store.state.auth).toEqual({
      status: 'not-authenticated',
      user: null,
      idToken: null,
      refreshToken: null
    })
  })
})
