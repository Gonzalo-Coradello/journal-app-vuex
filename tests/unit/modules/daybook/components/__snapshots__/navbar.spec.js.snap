// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar component > debe mostrar el componente correctamente 1`] = `[Function]`;
